base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -288,9 +285,13 @@ public IConnector getConnector(IConfigStore config)
             RemoteAuthority remauthority =
                     new RemoteAuthority(host, port, uri, timeout);
 
-            // Change end
-            if (nickname == null)
+            // Use client cert specified in KRA connector (i.e. subsystem cert)
+            String nickname = config.getString("nickName", null);
+            if (nickname == null) {
+                // Use CA signing cert as client certificate
                 nickname = mCA.getNickname();
```

Comment:
I know this came from old code, but  I don't think we should attempt to use ca signing cert as client cert.

Reply:
Agreed. I'd suggest to simplify the code as follows:
```
String nickname = config.getString("nickName");
```
The code will fail if `nickName` parameter is missing, so we're guaranteed to have a nickname.

Reply:
I dont' have a full view of the code, but if the exception is properly handled then yes.